### DISCUSSION TITLE
Companion app not gathering data from Dexcom one app

### DISCUSSION THREAD

**Discussion by Laura-Tanase** [NightscoutFoundation/xDrip#2571] [General]
I am trying to use the xdrip with the companion app feature to get the readings from my Dexcom one app (the original one not the BYODA) and I can't seem to get any readings no matter what I try.

All the notifications are fully enabled for the Dexcom one/Xdrip app and the permissions are also fully given for everything.  Battery use is set as unrestricted for both.

I also noticed that de notifications for the dexcom app does not show the BG number in i, just that the app is working. 

Thank you in advance :)

![321311644_1608322496287301_8921734777007255920_n](https://user-images.githubusercontent.com/54328293/208762985-7abdc410-58c6-451a-b626-61f8bc6f3100.jpg)
![Screenshot_20221220_214852_Settings](https://user-images.githubusercontent.com/54328293/208763002-80fdf4db-537b-4ff0-a5ed-9d1106fe6744.jpg)
![Screenshot_20221220_214806_Settings](https://user-images.githubusercontent.com/54328293/208763008-ff04fa3c-00d8-4e50-931e-25950dbfe6ed.jpg)
![Screenshot_20221220_215323_Settings](https://user-images.githubusercontent.com/54328293/208763026-50154ab1-6ee0-458c-b886-b158fe030355.jpg)

**Comment by jamorham** [comment]
Does the app have the "quick glance" feature? If so enable it.

**Reply by Laura-Tanase** [reply]
I finally have an answer to that from the dexcom support. it seems that the app does not have that feature. It still does not explain how some users can use it with xdrip companion app.

**Comment by Navid200** [comment]
@Laura-Tanase Is there a "Quick glance" feature?

Would you please tap on "Foreground Channel" on the screen you are showing above?

**Reply by Laura-Tanase** [reply]
it just manages weather the app shows any notifications at all

**Comment by SimonBoost** [comment]
I'm having this issue too. Just switched to ONE, started out using Xdrip natively, but am now using the BYOD Dexcom ONE app since they just updated it to the latest version 1.4.0 (needed to support the latest transmitters). I got one entry in the error log from UiBasedCollector saying "Receiving notifications that we are not enabled to process: com.dexcom.dexcomone. Nothing since. I too only have "Dexcom app is running" in the displayed notification, although that was also the case when I was using the G7 before switching, which worked perfectly with Xdrip, so that might be a red herring.

I'm running the 28th Jan daily build of Xdrip.

**Reply by Navid200** [reply]
Can you post a screenshot of one of the notifications?

**Reply by SimonBoost** [reply]
> Can you post a screenshot of one of the notifications?

I think the problem is with the Dexcom one app itself, as mentioned above it just doesn't seem to have a quick glance mode. Which makes it pretty useless. I'm now using it natively in xdrip, although that does mean no clarity upload. Plus I've had it drop out and ask to pair again twice on android 13 - hopefully I've found some settings to stop that now. Anyway, dexcom one notification is attached...
![Screenshot_20230131-131719](https://user-images.githubusercontent.com/109316045/215773007-0816da2a-e3ff-4db0-b1fb-d8a62541e325.png)

**Reply by Navid200** [reply]
That's the notification setting.  As long as there are no notifications including the blood glucose reading, xDrip Companion app mode will not work.  xDrip gets readings from those notifications in that mode.

If Clarity is a must, this is a solution usually:  https://navid200.github.io/xDrip/docs/DexcomAppxDrip.html
However, the last time I checked, BYODA does not have the ability to broadcast to xDrip neither for Dexcom One nor for G7.  This could be a reason to request from BYODA to add that ability for Dexcom One and G7.

**Reply by Navid200** [reply]
With respect to the connectivity issue using xDrip as the collector, please open a new discussion (not issue) here and let's have a look at what's going on.

**Reply by SimonBoost** [reply]
Yes, you're right, the BYODA version doesn't have broadcast currently. That's the version I'm running actually, since the official one only supports 3 year old phones you can no longer buy. I think the ONE app is completely different to the G6 so it's a lot of work to find out how to mod it, hence nobody has unfortunately. They did update the BYODA version to the latest one for me, because the older one didn't support transmitters beginning with C.

**Reply by SimonBoost** [reply]
> With respect to the connectivity issue using xDrip as the collector, please open a new discussion (not issue) here and let's have a look at what's going on.

Thanks, if it happens again I will do. I think it's an Android 13 thing, Bluetooth connectivity has been a bit dodgy ever since I upgraded.

**Comment by simondavies560** [comment]
Hi there. In case it is any help, I got the Dexcom G6 app to go from stating "Dexcom app is running" to showing the BG reading in the notification by turning the "Quick glance" setting off and then back on again.

However, now that I have the BG result showing in the Dexcom app notification, the xDrip+ companion app is still not getting results - just stating "BG Reading Unavailable" (as shown in earlier screenshots on this page). I have set all of the notifications and settings as suggested in instructions.

I'm using a Pixel 4a (4G) running Android 13, latest version of the Dexcom G6 app, and version e50dfdf-2-23.05.03 of xDrip.

Do you have any thoughts on anything else I can try to get this working?

Thanks
Simon

**Reply by Navid200** [reply]
This thread is about Dexcom ONE, not G6.
You could have started a new thread.

Anyway, these are the instructions: https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html

**Reply by simondavies560** [reply]
Apologies, my mistake. I have followed those instructions but with no luck. I will start a new thread, if that's preferable. Thanks

**Reply by Navid200** [reply]
Can you post a screenshot of one of the notifications?

**Reply by simondavies560** [reply]
Thanks for getting back to me Navid200. My daughter has just left for school with the phone in question. I will upload one as soon as I can. (I wasn't expecting such a prompt response, so thanks for that)

**Reply by Navid200** [reply]
The units shown  in the notification must match the units selected in xDrip.  If they don't match, it will not work.  That's what I wanted to see.  You can check that yourself, if you haven't already, when your daughter is back.

**Reply by simondavies560** [reply]
I will confirm later, but I am sure they are both set to mmol/l. Thanks

**Reply by simondavies560** [reply]
Hi again @Navid200 . I'm attaching several screenshots showing notifications, xDrip+ units, and the notification settings. Thanks

![Screenshot_20230724-182931](https://github.com/NightscoutFoundation/xDrip/assets/36174660/0527b248-f2a7-4882-a382-e0278f3f956f)
![Screenshot_20230724-183046](https://github.com/NightscoutFoundation/xDrip/assets/36174660/9ff70ed0-b35c-49fe-bae1-00a6ceef4c80)
![Screenshot_20230724-183148](https://github.com/NightscoutFoundation/xDrip/assets/36174660/c8351e75-a381-4215-8812-ef37a1c91be2)

**Reply by Navid200** [reply]
I need to see which version of xDrip you are using.
https://navid200.github.io/xDrip/docs/xDrip-Version.html

**Comment by simondavies560** [comment]
Version e50dfdf
<https://github.com/NightscoutFoundation/xDrip/commit/e50dfdfc4f6b47f9dc6448ebca392958e9286352>
-2-23.05.03

**Reply by Navid200** [reply]
Would you please create a new thread?
Please create a screenshot of the xDrip classic status page and include.
Please include the screenshots showing that xDrip has notification access. and that the units are the same.

Then, I can point the right deverloper to it.

**Reply by Navid200** [reply]
Also please include a screenshot of the hardware data source.

**Reply by simondavies560** [reply]
Will do, thank you!

**Reply by noisecontrol1** [reply]
Hi...did you solve that Problem...? my system worked perfect with a Samsung Note 10...camaps FX to xdrip with the companion mode ..now with a Pixel 7 pro...same settings...no Data to xdrip...thx..Arne

**Reply by simondavies560** [reply]
Hi @noisecontrol1 no I didn't ever get this sorted. I started a new thread (link below) but never solved it. I just set up nightscout to send the data straight to Dexcom Clarity rather than gathering it with xdrip.

https://github.com/NightscoutFoundation/xDrip/discussions/2961

**Reply by Navid200** [reply]
@simondavies560 How do you send data from Nightscout to Clarity?  Do you mind sharing the method?

**Reply by simondavies560** [reply]
Hi @Navid200 I use Heroku and set it up using the instructions in the link below, step 3 f)

https://nightscout.github.io/vendors/heroku/new_user/

**Reply by Navid200** [reply]
That's Dex share, right?
That's not Clarity.

@psonnera 
Can Nightscout upload to Clarity?

**Reply by simondavies560** [reply]
@Navid200 the instructions describe Dexcom Share but you can upload to Clarity. We have a Clarity account, use those credentials in Nightscout and get the data in Clarity.

**Reply by psonnera** [reply]
@simondavies560 as far as I know only the Dexcom app uploads to Clarity. Can you confirm your setup? No Dexcom app used at all?

**Reply by simondavies560** [reply]
Hey @Navid200 and @psonnera apologies, I was getting my nightscouts before my claritys. We do use Nightscout and clarity, but not how I described earlier.

I use the Dexcom G6 app to collect daughter's CGM data and send it to Clarity. I then have Nightscout set up to RECEIVE the data FROM clarity (as per Nightscout instructions), and use xdrip on my phone to access Nightscout. I have been playing around with different approaches because I find the Dexcom Follow app to be a bit tedious. 

So, my original answer to @noisecontrol1 should have stated that I never managed to get xdrip working as a companion app with Dexcom G6 app and so used Nightscout to gather data from Clarity and send to my xdrip.

**Comment by psonnera** [comment]
Report from Italy of ONE app version 1.5.0 (SW12585) showing notifications, xDrip+ latest Nightly doesn't catch them Unit set the same on both. 
![image](https://github.com/NightscoutFoundation/xDrip/assets/33809361/99bd9f5e-06a4-43e9-b8bc-c47f5ab0c143)

**Reply by Navid200** [reply]
I didn't think the Dexcom One app showed glucose notifications?
Can we see a sample?

**Reply by psonnera** [reply]
We checked with the requestor and this app only shows notifications when alerts trigger. i.e. cannot be used with xDrip+ companion app unfortunately.
